### PR TITLE
feat!: remove platform pinning

### PR DESCRIPTION
**Pull request by erisu**
### Platforms affected

none

### Motivation and Context

- Stop managing the platform pinning.
- Always allow the use of the latest release version without requiring an update of the lib or cli.

### Description

- Remove platform version pinning data.
- Remove logic for displaying the pinning version.
- Always fetch latest version when not supplied or already installed.
- Updated test specs to support code changes.

Previous concerns which shouldnt not be an issue anymore: Platforms relied on the common object which was passed to platform. Platforms should now be recreating the common object using its own installed common package.

E.g.
- [cordova-android](https://github.com/apache/cordova-android/blob/master/lib/Api.js#L118)
- [cordova-ios](https://github.com/apache/cordova-ios/blob/master/lib/Api.js#L213)
- [cordova-electron](https://github.com/apache/cordova-electron/blob/master/lib/Api.js#L96)

### Testing

In Progress

- [x] `npm t`
- [x] GH Actions

### Checklist

- [x] I've run the tests to see all new and existing tests pass
- [x] I added automated test coverage as appropriate for this change
- [ ] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [ ] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [ ] I've updated the documentation if necessary

### CODE REVIEW

**Comment by erisu** [comment]
Last call for review before merging.

File: src/cordova/platform/addHelper.js [hunk 1]
```diff
@@ -99,9 +99,9 @@ function addHelper (cmd, hooksRunner, projectRoot, targets, opts) {
                     }
 
                     // If spec still doesn't exist, try to use pinned version
```
Comment (breautek) [inline src/cordova/platform/addHelper.js:101]: nitpik: Comment still states attempting to use pinned version.

Comment (erisu) [reply src/cordova/platform/addHelper.js]: I updated the comment.

**Review by NiklasMerz — APPROVED** [review]
Code LGTM

**Review by breautek — APPROVED** [review]
lgtm

**Pull request merged**